3.evaluate_model/scripts/nbconverted/process_jump_phenotype_profiles.py

```diff
@@ -0,0 +1,283 @@
+# 
+# We then performed a series of KS tests to identify how different treatment distributions of all phenotype probabilities differed from controls.
+# 
+# See https://github.com/WayScience/JUMP-single-cell for complete details.
+# 
+# Here, we perform the following:
+# 
+# 1. Load in this data from the JUMP-single-cell repo
+# 2. Explore the top results per phenotype/treatment_type/model_type
+# 3. Convert it to wide format
+# 
+# This wide format represents a "phenotypic profile" which we can use similarly as an image-based morphology profile.
+# 
+# We also fit a UMAP to this phenotypic profile for downstream visualization.
+
+# In[1]:
+
+
+import pathlib
+import pandas as pd
+
+import umap
+
+
+# In[2]:
+
+
+# Set file paths
+# 1) JUMP phenotype probabilities from AreaShape model
+commit = "2c063b6dc48049201a57b060d18f97a5fc783488"
+
+url = "https://github.com/WayScience/JUMP-single-cell/raw"
+file = "3.analyze_data/class_balanced_well_log_reg_comparison_results/class_balanced_well_log_reg_areashape_model_comparisons.parquet"
+
+jump_sc_pred_file = f"{url}/{commit}/{file}"
+
+# 2) JUMP additional metadata needed to summarize/groupby results
+jump_metadta_commit = "a18fd7719c05b638c731142b0d42a92c645e2b33"
+
+jump_metadta_url = "https://github.com/jump-cellpainting/2023_Chandrasekaran_submitted/raw"
+jump_metadta_file = "benchmark/output/experiment-metadata.tsv"
+
+jump_metadata_full_file = f"{jump_metadta_url}/{jump_metadta_commit}/{jump_metadta_file}"
+
+# Set constants
+n_top_results_to_explore = 10
+
+
+# In[3]:
+
+
+# Set output files
+output_dir = "jump_phenotype_profiles"
+
+top_results_summary_file = pathlib.Path(output_dir, "jump_most_significant_phenotype_enrichment.tsv")
+final_jump_phenotype_file = pathlib.Path(output_dir, "jump_phenotype_profiles.tsv.gz")
+shuffled_jump_phenotype_file = pathlib.Path(output_dir, "jump_phenotype_profiles_shuffled.tsv.gz")
+
+jump_umap_file = pathlib.Path(output_dir, "jump_phenotype_profiling_umap.tsv.gz")
+
+
+# ## Load and process data
+
+# In[4]:
+
+
+# Load KS test results and drop uninformative columns
+jump_pred_df = (
+    pd.read_parquet(jump_sc_pred_file)
+    .drop(columns=["statistical_test", "comparison_metric"])
+)
+
+print(jump_pred_df.shape)
+jump_pred_df.head()
+
+
+# In[5]:
+
+
+# Load JUMP metadata for JUMP-CP Pilot
+# For an explanation of these metadata columns see: 
+# https://github.com/jump-cellpainting/2023_Chandrasekaran_submitted/blob/9edd26d60524a62f993d4df40a5d8908206714f5/README.md#batch-and-plate-metadata
+jump_metadata_df = (
+    pd.read_csv(jump_metadata_full_file, sep="\t")
+    .query("Batch == '2020_11_04_CPJUMP1'")
+)
+
+print(jump_metadata_df.shape)
+jump_metadata_df.head()
+
+
+# In[6]:
+
+
+# Merge dataframes and retain only informative columns
+jump_pred_df = (
+    jump_pred_df
+    .merge(
+        jump_metadata_df,
+        left_on="Metadata_Plate",
+        right_on="Assay_Plate_Barcode"
+    )
```

Comment:
Yes, I think it makes sense to use this version now - thanks for the pointer!

BTW, I took a look at your most recent PR to add this info. Nice work! I would also recommend adding a pointer indicating the provenance of the `experiment-metadata.tsv` https://github.com/WayScience/JUMP-single-cell/pull/21/files#diff-6f3ca646908f89153386be951563a61449e8f5df46549d224ff98b74d6aab859

You currently include this file in the repo, but someone might not know this files origin. I'm adding my details below (I'll remove then in the next commit) in case you decide to incorporate it in the JUMP-single-cell repo in some form (maybe in a README as a quick note)

```python
# 2) JUMP additional metadata needed to summarize/groupby results
jump_metadata_commit = "a18fd7719c05b638c731142b0d42a92c645e2b33"

jump_metadata_url = "https://github.com/jump-cellpainting/2023_Chandrasekaran_submitted/raw"
jump_metadata_file = "benchmark/output/experiment-metadata.tsv"

# Load JUMP metadata for JUMP-CP Pilot
# For an explanation of these metadata columns see: 
# https://github.com/jump-cellpainting/2023_Chandrasekaran_submitted/blob/9edd26d60524a62f993d4df40a5d8908206714f5/README.md#batch-and-plate-metadata
jump_metadata_df = (
    pd.read_csv(jump_metadata_full_file, sep="\t")
    .query("Batch == '2020_11_04_CPJUMP1'")
)

print(jump_metadata_df.shape)
jump_metadata_df.head()
```

Reply:
Also, I noticed that there are 720 additional rows in the new `JUMP-single-cell` parquet file. I couldn't track down why, but it probably has to do with my merge somehow dropping columns. I don't think there's anything to do with this info, just briefly noting here.